json.go

```diff
@@ -0,0 +1,373 @@
+// Copyright 2015 The go-ethereum Authors
```

Comment:
File taken straight from the go-ethereum repository - do not review

Reply:
Why not use this package from upstream? Why does it need to be checked in

Reply:
I need a lot of its internals that are unexported variables and methods